mobile_app/lib/view/cafe_detail/cafe_detail_body_content.dart

```diff
@@ -0,0 +1,129 @@
+import 'package:flutter/material.dart';
+import 'package:mobile_app/constants/type.dart';
+import 'package:mobile_app/util/cafe_detail.dart';
+import 'package:mobile_app/view/cafe_detail/cafe_detail_footer.dart';
+import 'package:mobile_app/view/cafe_detail/cafe_detail_info.dart';
+import 'package:mobile_app/view/cafe_detail/cafe_detail_location.dart';
+import 'package:mobile_app/view/cafe_detail/cafe_detail_menu.dart';
+import 'package:mobile_app/view/common/cafe_image_slider.dart';
+import 'package:mobile_app/view/common/image_index_bullet.dart';
+
+const double FOOTER_HEIGHT = 72;
+
+class DetailBodyContent extends StatefulWidget {
+  final CafeModel cafe;
+
+  DetailBodyContent({required this.cafe});
+
+  @override
+  _DetailBodyContentState createState() => _DetailBodyContentState(cafe: cafe);
+}
+
+class _DetailBodyContentState extends State<DetailBodyContent> {
+  final CafeModel cafe;
+  final PageController _controller = PageController();
+  final GlobalKey menuKey = new GlobalKey();
+  final GlobalKey mapKey = new GlobalKey();
+  bool hasLocation = false;
+  bool hasMenus = false;
+  double footerOffset = 0;
+  int? currentIndex;
+
+  _DetailBodyContentState({required this.cafe});
+
+  @override
+  void initState(){
+    super.initState();
+    hasLocation = hasCafeMetadata(cafe.metadata?.location?.lat) &&
+        hasCafeMetadata(cafe.metadata?.location?.lng);
+    hasMenus =  (cafe.metadata?.menus ?? []).length > 0;
+    footerOffset = (hasLocation || hasMenus) ? -FOOTER_HEIGHT : 0;
+  }
+
+  void handleImageSlide(int index) {
+    setState(() {
+      currentIndex = index % cafe.image.count;
+    });
+  }
+  void handleScroll(GlobalKey targetKey) {
+    Scrollable.ensureVisible(targetKey.currentContext!,
+        duration: Duration(milliseconds: 500),
+        curve: Curves.easeInOut
+    );
+  }
```

Comment:
nullable한 값을 assertion으로 넘기기보다는 적절하게 예외처리를 해주세요.

```suggestion
  void handleScroll(GlobalKey targetKey) {
    final context = targetKey.currentContext;
    if (context == null) { return }
  
    Scrollable.ensureVisible(context,
        duration: Duration(milliseconds: 500),
        curve: Curves.easeInOut
    );
  }
```